c2/c2-client-bundle/c2-client-service/src/main/java/org/apache/nifi/c2/client/service/operation/DebugOperationHandler.java

```diff
@@ -0,0 +1,257 @@
+        if (debugCallbackUrl == null) {
+            LOG.error("Callback URL was not found in C2 request.");
+            return operationAckCreate.apply(operationState(NOT_APPLIED, C2_CALLBACK_URL_NOT_FOUND));
+        }
+
+        List<Path> bundleFilePaths = concat(
+            debugBundlePaths(logDir, LogFile.values()),
+            debugBundlePaths(configDir, ConfigFile.values()))
+            .collect(toList());
+
+        List<Path> contentFilteredFilePaths = null;
+        C2OperationState operationState;
+        try {
+            contentFilteredFilePaths = filterContent(bundleFilePaths);
+            operationState = createDebugBundle(contentFilteredFilePaths)
+                .map(bundle -> c2Client.uploadDebugBundle(debugCallbackUrl, bundle)
+                    .map(errorMessage -> operationState(NOT_APPLIED, errorMessage))
+                    .orElseGet(() -> operationState(FULLY_APPLIED, SUCCESSFUL_UPLOAD)))
+                .orElseGet(() -> operationState(NOT_APPLIED, UNABLE_TO_CREATE_BUNDLE));
+        } catch (Exception e) {
+            LOG.error("Unexpected error happened", e);
+            operationState = operationState(NOT_APPLIED, UNABLE_TO_CREATE_BUNDLE);
+        } finally {
+            ofNullable(contentFilteredFilePaths).ifPresent(this::cleanup);
+        }
+
+        LOG.debug("Returning operation ack with state={} and details={}", operationState.getState(), operationState.getDetails());
+        return operationAckCreate.apply(operationState);
+    }
+
+    private Function<C2OperationState, C2OperationAck> operationAck(C2Operation operation) {
+        return state -> {
+            C2OperationAck operationAck = new C2OperationAck();
+            operationAck.setOperationId(ofNullable(operation.getIdentifier()).orElse(EMPTY));
+            operationAck.setOperationState(state);
+            return operationAck;
+        };
+    }
+
+    private C2OperationState operationState(OperationState operationState, String details) {
+        C2OperationState state = new C2OperationState();
+        state.setState(operationState);
+        state.setDetails(details);
+        return state;
+    }
+
+    private Stream<Path> debugBundlePaths(String baseDir, DebugBundleFile... debugBundleFiles) {
+        return Stream.of(debugBundleFiles)
+            .map(bundleFile -> Paths.get(baseDir, bundleFile.getFileName()))
+            .filter(Files::exists)
+            .filter(Files::isRegularFile);
+    }
+
+    private List<Path> filterContent(List<Path> bundleFilePaths) {
+        List<Path> contentFilteredFilePaths = new ArrayList<>();
+        for (Path path : bundleFilePaths) {
+            String fileName = path.getFileName().toString();
+            try {
+                Path tempDirectory = createTempDirectory(null);
+                Path tempFile = Paths.get(tempDirectory.toAbsolutePath().toString(), fileName);
+                write(tempFile, (Iterable<String>) lines(path).filter(contentFilter)::iterator);
```

Comment:
I think lines() needs to be used with try with resource

Reply:
Thanks for spotting this, I missed that